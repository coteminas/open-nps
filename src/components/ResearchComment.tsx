import React from 'react';

import TextField from '@material-ui/core/TextField';

export const ResearchComment = ({ value, setValue }) => (
  <TextField
    multiline
    variant="outlined"
    onChange={(e) => { setValue(e.target.value) }}
    value={value}
  />
)
