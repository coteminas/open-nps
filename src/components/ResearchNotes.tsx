import React from 'react';

import Button from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import { withStyles, WithStyles, createStyles } from '@material-ui/styles';

/* istanbul ignore next */
const styles = (theme) =>
  createStyles({
    buttonDiv: {
      maxWidth: '700px',
      width: '100%',
      display: 'flex',
      justifyContent: 'space-around',
      flexWrap: 'wrap',

      '& > *': {
        marginBottom: theme.spacing(1),
      },
    },
  });

interface Props extends WithStyles<typeof styles> {
  themeOpts: ThemeOptionsConfigValues;
  selected?: number;
  setValue: SimpleFn<string, void>;
}

const renderNotes = (
  themeOpts: ThemeOptionsConfigValues,
  selected,
  setValue
): React.ReactNodeArray =>
  Object.keys(new Array(10).fill(1)).map((val) => {
    const note = parseInt(val) + 1;
    return (
      <Button
        key={`nps-${val}`}
        onClick={() => setValue(note)}
        size={themeOpts.ResearchNotesBtnSize}
        variant={note === selected ? 'contained' : 'outlined'}
        color={themeOpts.ResearchNotesBtnColor}
        data-cy="ResearchNoteBtn"
      >
        {note}
      </Button>
    );
  });

export const ResearchNotes: React.FC<Props> = ({
  themeOpts,
  selected,
  setValue,
  classes,
}) => {
  const noteElementsArr = renderNotes(themeOpts, selected, setValue);

  if (!themeOpts.ResearchNotesBtnGroupActive) {
    return <div className={classes.buttonDiv}>{noteElementsArr}</div>;
  }

  return <ButtonGroup>{noteElementsArr}</ButtonGroup>;
};

export default withStyles(styles)(ResearchNotes);
