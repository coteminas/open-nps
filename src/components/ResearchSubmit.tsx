import React from 'react';

import MUIButton from '@material-ui/core/Button';
import { withStyles } from '@material-ui/styles';

interface Props {
  children: React.ReactChild;
  themeOpts: ThemeOptionsConfigValues;
}

const Button = withStyles((theme) => ({
  root: {
    marginTop: theme.spacing(3),
    width: '600px',
  },
}))(MUIButton);

export const ResearchSubmit: React.FC<Props> = ({ themeOpts, children }) => (
  <Button
    color={themeOpts.ResearchSubmitBtnColor}
    variant="contained"
    type="submit"
  >
    {children}
  </Button>
);

export default ResearchSubmit;
