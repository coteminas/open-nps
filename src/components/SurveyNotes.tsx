import React from 'react';

import MUIButton from '@material-ui/core/Button';
import ButtonGroup from '@material-ui/core/ButtonGroup';
import { withStyles, WithStyles, createStyles } from '@material-ui/styles';
import { ThemeOptions } from '@material-ui/core';

/* istanbul ignore next */
export const Button = withStyles((theme: ThemeOptions) => ({
  root: {
    minWidth: '35px',
    fontSize: '16px',
  },
}))(MUIButton);

/* istanbul ignore next */
const styles = (theme) =>
  createStyles({
    buttonDiv: {
      maxWidth: '620px',
      width: '100%',
      display: 'flex',
      justifyContent: 'space-around',
      flexWrap: 'wrap',

      '& > *': {
        marginBottom: theme.spacing(1),
      },
    },
  });

interface Props extends WithStyles<typeof styles> {
  themeOpts: ThemeOptionsConfigValues;
  selected?: number;
  setValue: SimpleFn<string, void>;
}

const renderNotes = (
  themeOpts: ThemeOptionsConfigValues,
  selected,
  setValue
): React.ReactNodeArray =>
  Object.keys(new Array(11).fill(1)).map((val) => {
    const note = parseInt(val);
    return (
      <Button
        key={`nps-${val}`}
        onClick={() => setValue(note)}
        size={themeOpts.SurveyNotesBtnSize}
        variant={note === selected ? 'contained' : 'outlined'}
        color={themeOpts.SurveyNotesBtnColor}
        data-cy="SurveyNoteBtn"
      >
        {note}
      </Button>
    );
  });

export const SurveyNotes: React.FC<Props> = ({
  themeOpts,
  selected,
  setValue,
  classes,
}) => {
  const noteElementsArr = renderNotes(themeOpts, selected, setValue);

  if (!themeOpts.SurveyNotesBtnGroupActive) {
    return <div className={classes.buttonDiv}>{noteElementsArr}</div>;
  }

  return <ButtonGroup>{noteElementsArr}</ButtonGroup>;
};

export default withStyles(styles)(SurveyNotes);
