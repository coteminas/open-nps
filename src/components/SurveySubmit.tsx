import React from 'react';

import MUIButton from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import { withStyles, createStyles } from '@material-ui/styles';

interface Props {
  children: React.ReactChild;
  themeOpts: ThemeOptionsConfigValues;
  isSending: boolean;
}

/* istanbul ignore next */
const btnStyles = (theme) =>
  createStyles({
    root: {
      marginTop: theme.spacing(3),
      maxWidth: '600px',
    },
  });

const Button = withStyles(btnStyles)(MUIButton);

export const SurveySubmit: React.FC<Props> = ({
  isSending,
  themeOpts,
  children,
}) => (
  <Button
    color={themeOpts.SurveySubmitBtnColor}
    variant="contained"
    type="submit"
    data-cy="SurveySubmit"
    fullWidth={themeOpts.SurveySubmitBtnFullWidth}
    disabled={isSending}
  >
    {isSending ? (
      <CircularProgress size={themeOpts.SurveySubmitBtnLoadSize} />
    ) : (
      children
    )}
  </Button>
);

export default SurveySubmit;
