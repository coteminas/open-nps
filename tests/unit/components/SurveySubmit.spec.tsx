import React from 'react';

import { shallow } from 'enzyme';

import { SurveySubmit } from '~/components/SurveySubmit';

describe('/src/components/SurveyNotes', () => {
  const baseProps = {
    themeOpts: {
      SurveySubmitBtnColor: 'primary',
    } as ThemeOptionsConfigValues,
    children: 'FooBar',
  };

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render properly', () => {
    const wrap = shallow(<SurveySubmit {...baseProps} />);

    expect(wrap).toHaveProp('color', baseProps.themeOpts.SurveySubmitBtnColor);
    expect(wrap).toHaveProp('variant', 'contained');
    expect(wrap).toHaveProp('type', 'submit');
    expect(wrap).toHaveText(baseProps.children);
  });
});
