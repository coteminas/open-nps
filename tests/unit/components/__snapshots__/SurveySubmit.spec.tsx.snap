// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/src/components/SurveyNotes should render properly 1`] = `
<WithStyles(WithStyles(ForwardRef(Button)))
  color="primary"
  data-cy="SurveySubmit"
  disabled={false}
  type="submit"
  variant="contained"
>
  FooBar
</WithStyles(WithStyles(ForwardRef(Button)))>
`;

exports[`/src/components/SurveyNotes should render properly with isSending === true 1`] = `
<WithStyles(WithStyles(ForwardRef(Button)))
  color="primary"
  data-cy="SurveySubmit"
  disabled={true}
  type="submit"
  variant="contained"
>
  <WithStyles(ForwardRef(CircularProgress)) />
</WithStyles(WithStyles(ForwardRef(Button)))>
`;
